dotnet/test/AutoGen.Anthropic.Tests/AnthropicClientTest.cs

```diff
@@ -165,6 +174,60 @@ public async Task AnthropicClientTestToolChoiceAsync()
         Assert.True(toolUseContent.Input is JsonNode);
     }
 
+    [ApiKeyFact("ANTHROPIC_API_KEY")]
+    public async Task AnthropicClientChatCompletionCacheControlTestAsync()
+    {
+        var anthropicClient = new AnthropicClient(new HttpClient(), AnthropicConstants.Endpoint, AnthropicTestUtils.ApiKey);
+
+        var request = new ChatCompletionRequest();
+        request.Model = AnthropicConstants.Claude35Sonnet;
+        request.Stream = false;
+        request.MaxTokens = 100;
+
+        request.SystemMessage =
+        [
+            SystemMessage.CreateSystemMessageWithCacheControl(
+                $"You are an LLM that is great at remembering stories {AnthropicTestUtils.LongStory}"),
+        ];
+
+        request.Messages =
+        [
+            new ChatMessage("user", "What should i know about Bob?")
+        ];
+
+        var response = await anthropicClient.CreateChatCompletionsAsync(request, CancellationToken.None);
+        response.Usage.Should().NotBeNull();
+
+        // There's no way to clear the cache. Running the assert frequently may cause this to fail because the cache is already been created 
+        // response.Usage!.CreationInputTokens.Should().BeGreaterThan(0);
+        // The cache reduces the input tokens. We expect the input tokens to be less the large system prompt and only the user message
+        response.Usage!.InputTokens.Should().BeLessThan(20);
+
+        request.Messages =
+        [
+            new ChatMessage("user", "Summarize the story of bob")
+        ];
+
+        response = await anthropicClient.CreateChatCompletionsAsync(request, CancellationToken.None);
+        response.Usage.Should().NotBeNull();
+        response.Usage!.ReadInputTokens.Should().BeGreaterThan(0);
+        response.Usage!.InputTokens.Should().BeLessThan(20);
+
+        // Should not use the cache
+        request.SystemMessage =
+        [
+            SystemMessage.CreateSystemMessage("You are a helpful assistant.")
+        ];
+
+        request.Messages =
+        [
+            new ChatMessage("user", "What are some text editors I could use to write C#?")
+        ];
+
+        response = await anthropicClient.CreateChatCompletionsAsync(request, CancellationToken.None);
+        response.Usage!.ReadInputTokens.Should().Be(0);
```

Comment:
Why this should be 0, shouldn't the cache not being hit?

Reply:
We don't expect the cache to be hit / used because we changed the system message.
there is nothing to reuse in the cache, so we expect the `ReadInputTokens` to be 0.